File: Scripts/micro_dao/cidade.py
from micro_dao.objeto_modelo import ObjetoModelo

class Cidade(ObjetoModelo):


    _id: int
    _nome: str
    _fk_estado: int

    def __init__(self, nome_cidade: str, fk_estado: int):
        self.set_nome(nome_cidade)
        self.set_fk_estado(fk_estado)

    def get_id(self) -> int:
        return self._id

    def set_id(self, id: int):
        self._id = id
        return self

    def get_nome(self) -> str:
        return self._nome

    def set_nome(self, nome_cidade: str):
        self._nome = nome_cidade
        return self

    def get_fk_estado(self) -> int:
        return self._fk_estado

    def set_fk_estado(self, fk_estado: int):
        self._fk_estado = fk_estado
        return self

File: Projeto_Java/src/Dominio/Entidades/Cidade.java
package Dominio.Entidades;

import Dominio.Enum.EEstado;

import javax.persistence.*;

@Entity
@Table(name = "cidade")
public class Cidade {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "id")
    private int id;

    @Column(name = "nome", length = 40)
    private String nome;

    @Column(name = "fk_estado")
    private int fkEstado;

    private EEstado estado;


    public Cidade() {}

    public Cidade(String nome, int fkEstado){
        setNome(nome);
        setFkEstado(fkEstado);
    }

    public Cidade(int id, String nome, int fkEstado){
        setId(id);
        setNome(nome);
        setFkEstado(fkEstado);
    }


    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    public String getNome() {
        return nome;
    }

    public void setNome(String nome) {
        this.nome = nome;
    }

    public int getFkEstado() {
        return fkEstado;
    }

    public void setFkEstado(int fkEstado) {
        this.fkEstado = fkEstado;
        this.setEstado();
    }

    @PostLoad
    private void setEstado() {
        this.estado = EEstado.getById(this.fkEstado);
    }

    public EEstado getEstado() {
        return estado;
    }
}
